scripts/githooks/commit_msg.py

```diff
@@ -0,0 +1,183 @@
+              'Limit the subject line to 50 characters',
+              'Capitalize the subject line',
+              'Do not end the subject line with a period',
+              'Use the imperative mood in the subject line',
+              'Wrap the body at 72 characters',
+              'Use the body to explain what and why vs. how' ]
+
+    valid_rules = [ False, False, False, False, False, False, False ]
+
+    def __init__(self, filename = None):
+        if filename != None:
+            with open(filename, 'r') as f:
+                self.parse_lines(f.readlines())
+
+    def parse_lines(self, lines):
+        self.title = lines[0]
+        self.title_words = self.title.split()
+        self.has_subject_body_separator = False
+        self.body_lines = []
+
+        if len(lines) > 1:
+            self.has_subject_body_separator = (len(lines[1].strip()) == 0)
+
+            if self.has_subject_body_separator:
+                self.body_lines = lines[2:]
+            else:
+                self.body_lines = lines[1:]
+
+        return self
+
+    def check_subject_body_separtor(self):
+        'Rule 1: Separate subject from body with a blank line'
+
+        if len(self.body_lines) > 0:
+            return self.has_subject_body_separator
+        return True
+
+    def check_subject_limit(self):
+        'Rule 2: Limit the subject line to 50 characters'
+        return len(self.title) < 50
+
+    def check_subject_capitalized(self):
+        'Rule 3: Capitalize the subject line'
+        return len(self.title) > 0 and self.title[0].isupper()
+
+    def check_subject_no_period(self):
+        'Rule 4: Do not end the subject line with a period'
+        return not self.title.endswith('.')
+
+    common_first_words = [ 'Add',
+                           'Adjust',
+                           'Support',
+                           'Change',
+                           'Remove',
+                           'Fix',
+                           'Print',
+                           'Track',
+                           'Refactor',
+                           'Combine',
+                           'Release',
+                           'Set',
+                           'Stop',
+                           'Make',
+                           'Enable',
+                           'Check',
+                           'Exclude']
+
+    def check_subject_imperative(self):
+        'Rule 5: Use the imperative mood in the subject line.'
+        'We can only check for common mistakes here, like using'
+        'the -ing form of a verb or non-imperative version of '
+        'common verbs'
+
+        firstword = self.title_words[0]
+
+        if firstword.endswith('ing'):
+            return False
+
+        for word in self.common_first_words:
+            if firstword.startswith(word) and firstword != word:
+                return False
+
+        return True
+
+    def check_body_limit(self):
+        'Rule 6: Wrap the body at 72 characters'
```

Comment:
never got why this wasn't 80, but because github does sometimes render this weirdly I don't mind it. Certainly don't mind as much as the 50 char limit on title

Reply:
I think we should stick to following the seven rules listed, since it makes things easy and we can point to the blog without exceptions. IMO, I don't think it is worth debating a few characters back-and-forth.

Reply:
(git log indents, so if you use the entire terminal width, it won't render right)

Reply:
yeah, I think there a number of reasons for these limits, as explained in the blog post. I think we should try sticking to the rules for now and see how it works out. Once we're used to these rules, I am confident that it won't be such a big issue, IMHO.